api/v1beta1/azuremachine_types.go

```diff
@@ -115,7 +116,8 @@ type AzureMachineSpec struct {
 	// +optional
 	SecurityProfile *SecurityProfile `json:"securityProfile,omitempty"`
 
-	// SubnetName selects the Subnet where the VM will be placed
+	// SubnetName (deprecated) selects the Subnet where the VM will be placed.
+	// This field is deprecated. The subnetName should be set in the networkInterfaces field.
```

Comment:
```suggestion
	// // Deprecated: subnetName should be set in the networkInterfaces field.
```

to be consistent with other deprecated fields in the API

Reply:
ditto for others